cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/profile/ProfileMetadataMessageProcessor.java

```diff
@@ -365,4 +377,41 @@ private void emitApplicationCountMetric() {
       LOG.warn("Failed to get application count", e);
     }
   }
+
+  private void emitPluginCountMetric() {
+    emitPluginCountMetric(null);
+  }
+
+  private void emitPluginCountMetric(Set<ImmutablePair<String, String>> deletedPlugins) {
+    try {
+      List<ImmutablePair<String, String>> remainingPlugins = appMetadataStore.getAllApplications().stream()
```

Comment:
This seems like an expensive call, how often are we doing this?

Reply:
Hmm, seems like we do this every time we create/delete a pipeline? Do we know how much this affects create/delete in instances with many pipelines?